File: lib/models/appModels/SocialMediaApp.dart
import '../enums/SocialMediaType.dart';
import 'User.dart';
import 'Team.dart';
import 'Event.dart';

class SocialMediaApp {
  SocialMediaType? type;
  String? url;
  User? user;
  Team? team;
  Event? event;

  SocialMediaApp({
    this.type,
    this.url,
    this.user,
    this.team,
    this.event,
  });
}

File: apollo/src/SocialMediaApp.ts
import { SocialMediaType } from "./enums";

import mongoose from 'mongoose';
const Schema = mongoose.Schema;


const socialMediaAppSchema = new mongoose.Schema({
    type: SocialMediaType,
    url: String,
    user: { type: Schema.Types.ObjectId, ref: 'User' },    
    event: { type: Schema.Types.ObjectId, ref: 'Event' },    
    fieldLocation: { type: Schema.Types.ObjectId, ref: 'FieldLocation' },
});

const SocialMediaApp = mongoose.model('SocialMediaApp', socialMediaAppSchema);

export default SocialMediaApp;